src/sentry/api/endpoints/organization_details.py

```diff
@@ -564,22 +567,24 @@ def handle_delete(self, request: Request, organization):
             return self.respond({"detail": ERR_DEFAULT_ORG}, status=400)
 
         with transaction.atomic():
-            updated = Organization.objects.filter(
-                id=organization.id, status=OrganizationStatus.ACTIVE
-            ).update(status=OrganizationStatus.PENDING_DELETION)
-            if updated:
-                organization.status = OrganizationStatus.PENDING_DELETION
+            updated_organization = mark_organization_as_pending_deletion_with_outbox_message(
```

Comment:
I wonder if we should use these functions as the start of a domain layer for organization operations? Several operations for organizations are collecting more behavior that needs to be used as a unit.